backend/prompt_studio/prompt_studio_core/views.py

```diff
@@ -44,29 +69,30 @@ class PromptStudioCoreView(viewsets.ModelViewSet):
 
     versioning_class = URLPathVersioning
 
-    permission_classes = [IsOwner]
     serializer_class = CustomToolSerializer
 
     processor_plugins = load_plugins()
 
+    def get_permissions(self) -> list[Any]:
+        if self.action == "destroy":
+            return [IsOwner()]
+
+        return [IsOwnerOrSharedUser()]
+
     def get_queryset(self) -> Optional[QuerySet]:
         filter_args = FilterHelper.build_filter_args(
             self.request,
             ToolStudioKeys.CREATED_BY,
         )
         if filter_args:
-            queryset = CustomTool.objects.filter(
-                created_by=self.request.user, **filter_args
+            queryset = CustomTool.objects.for_user(self.request.user).filter(
```

Comment:
@johnyrahul we could simplify it by adding this as a method to the model [like so](https://stackoverflow.com/a/59649302/12864260)

Reply:
Confused!.. That is what I have done here. @chandrasekharan-zipstack 

I think the if else is not required will remove them

Reply:
I think the else is okay here, it'll not be needed if we specify a default `queryset` as a class instance